crates/dids/src/method/dht/document_packet/service.rs

```diff
@@ -0,0 +1,222 @@
+use std::collections::HashMap;
+
+use crate::document::Service;
+
+use simple_dns::{
+    rdata::{RData, TXT},
+    Name, ResourceRecord,
+};
+
+use super::{DocumentPacketError, DEFAULT_TTL};
+
+/// Gets the RData from the record. If RData is RData::TXT, get the text as a string.
+/// Convert strings like "id=foo;t=bar;se=baz" into a map like { 'id': 'foo', 't': 'bar', 'se': 'baz' }
+/// If there is any issue, return DocumentPacketError::RDataError
+fn record_rdata_to_hash_map(
+    record: ResourceRecord,
+) -> Result<HashMap<String, String>, DocumentPacketError> {
+    // Get RData text as String
+    let rdata_txt = match record.rdata {
+        RData::TXT(txt) => txt,
+        _ => {
+            return Err(DocumentPacketError::RDataError(
+                "RData must have type TXT".to_owned(),
+            ))
+        }
+    };
+    let text = match String::try_from(rdata_txt) {
+        Ok(text) => text,
+        Err(_) => {
+            return Err(DocumentPacketError::RDataError(
+                "Failed to convert to string".to_owned(),
+            ))
+        }
+    };
+
+    // Parse key-value pairs:
+    //   Split string by ";" to get entries
+    //   Split each entry by "=" to get key and value
+    let mut attributes = HashMap::new();
+    for entry in text.split(';') {
+        let k_v: Vec<&str> = entry.split('=').collect();
+        if k_v.len() != 2 {
+            return Err(DocumentPacketError::RDataError(
+                "Could not get values from RData text".to_owned(),
+            ));
+        }
+
+        let k = k_v[0].trim().to_string();
+        let v = k_v[1].trim().to_string();
```

Comment:
don't we also have to delimit the k_v[1] by `,`? or do we just store the values as string here

example: `v=0;vm=k0,k1,k2;auth=k0;asm=k1;inv=k2;del=k2;svc=s0,s1,s2`

vm=k0,k1,k2 should be split into `{ vm: [k0, k1, k2] } ` or is it `{ vm: "k0,k1,k2" }`?

Reply:
<img width="840" alt="image" src="https://github.com/TBD54566975/web5-rs/assets/118510103/41c4db7e-238f-4591-a4c6-03b3a8074c94">

relevant section in did dht spec abt service https://did-dht.com/#services